core/templates/dev/head/pages/exploration_player/ConversationSkinDirective.js

```diff
@@ -912,6 +955,16 @@ oppia.directive('conversationSkin', [
           // of ConversationSkinDirective.
           $scope.interactionAnswerIsValid = true;
           $scope.setInteractionAnswerValidity = function(answerValidity) {
+            var currentIndex = PlayerPositionService.getActiveCardIndex();
+            // This check is added because it was observed that when returning
```

Comment:
Good explanation!

The guard added here looks OK. But _why_ is the Submit button sometimes falsely disabled? It seems like we should get to the bottom of that.

Reply:
I tried to track it, and it seems to happen when the left arrow is clicked on a concept card (which is in the history). So, if the learner is on an interaction, goes back (clicking left arrow on a concept card along the way) and  comes back, the submit button was disabled.
Also, this happened only in graph input interaction, so this might be restricted to supplementary cards. Will see.

Reply:
We should resolve this before merging this PR.

Reply:
Yeah, found out the issue. The problem was isConceptCardPending() function which was passed to tutor card directive had a delay in actually hiding the interaction. So, when a concept card was pending, the interaction (with blank input) was created and then hidden, but this shouldn't be created in the first place. Because of this, this card would have an interaction with invalid input always, just that it was hidden. So, whenever the learner navigated to this card, answerIsValid was set to false which in the end affected the interaction the learner was currently in.

This doesn't happen when for example a correct answer is given and a feedback is displayed (normal playthrough), because there is a check for activeCard.destStateName (which is set when a normal pending card is present) also for this interaction div, which happens synchronously and doesn't need a function being passed through directive attributes.

One way to fix it is to leave the above created check as it is.
Another way is to manually give some predefined constant as the destination state name for when a concept card is pending. That would also work. Also, when checking the transcript later on, that might make it more understandable as now it is as if destStateName is null and yet a concept card was displayed.

Reply:
For now, I kept this check and added a hardcoded destination called 'conceptCard' when a concept card is pending. If this method is fine, will add a global constant for this, with a different name?

Reply:
So I'm not sure I understand what's going on here. Both fixes seem a bit hacky. 

Shouldn't we avoid even creating the bad interaction in the first place (or perhaps replacing it with something like Continue or similar)? Or, in the place where we check for activeCard.destStateName, also include a check for something like "conceptCardIsNowPending", since presumably the "destination" of the student has been decided by then?

Passing functions through directive attrs shouldn't lead to synchronicity issues. Is there any part of this pipeline that's actually async (due to timeouts, etc.)? If not, then there shouldn't be issues with synchronicity.

Finally, there probably ought to be a separate field in activeCard for "leadsToConceptCard". But I don't think we should invent special destStateNames that are meant to indicate a concept card -- there is a possible clash with actual state names within explorations.

So, overall -- good job on finding the root cause! But we need a fix that actually addresses that head-on and doesn't introduce muddy concepts.

Reply:
Ok, yeah, adding the leadsToConceptCard field to activeCard would fix the issue also.
Shall I do that?
Also, I am not sure why when I passed through directives, it didn't work. Also, I can confirm there is a time delay since the first time when I played the exploration and had a concept card pending I could briefly see the interaction which then hides and the continue button becomes visible. 
The above fix of adding the field would prevent that also. Shall I do it?

Reply:
Sure, that sounds good. Just make sure that that field is in activeCard right from the start, and that you're not adding the field to a dict later where there was no field before.

Reply:
Ok, yeah, will add the field everywhere.
Also, when the refactor to objects are done, should this field also be added to card object? or to transcript elements?

Reply:
Done, added the leadsToConceptCard field.